tests/iface_namingmode/test_iface_namingmode.py

```diff
@@ -526,17 +535,19 @@ def test_show_queue_counters(self, setup, setup_config_mode, duthosts, enum_rand
         if mode == 'alias':
             for intf in interfaces:
                 alias = setup['port_name_map'][intf]
-                assert (re.search(r'{}\s+[U|M]C|ALL\d\s+\S+\s+\S+\s+\S+\s+\S+'
-                                  .format(alias), queue_counter) is not None) \
-                    and (setup['port_alias_map'][alias] not in queue_counter)
+                assert (re.search(QUEUE_COUNTERS_RE_FMT.format(alias),
+                                  queue_counter) is not None) \
+                    and (re.search(QUEUE_COUNTERS_RE_FMT.format(setup['port_alias_map'][alias]),
+                                   queue_counter) is None)
                 intfsChecked += 1
         elif mode == 'default':
             for intf in interfaces:
                 if intf not in setup['port_name_map']:
                     continue
-                assert (re.search(r'{}\s+[U|M]C|ALL\d\s+\S+\s+\S+\s+\S+\s+\S+'
-                                  .format(intf), queue_counter) is not None) \
-                    and (setup['port_name_map'][intf] not in queue_counter)
+                assert (re.search(QUEUE_COUNTERS_RE_FMT.format(intf),
+                                  queue_counter) is not None) \
+                    and (re.search(QUEUE_COUNTERS_RE_FMT.format(setup['port_name_map'][intf]),
+                                   queue_counter) is None)
```

Comment:
There can be a case where a SONiC interface name is the same as Alias name in certain HwSKUs.
For example, https://github.com/sonic-net/sonic-buildimage/blob/master/device/arista/x86_64-arista_7050_qx32s/Arista-7050QX32S-Q32/port_config.ini#L29 here, Ethernet32 is present in alias and SONiC interface.
The test will fail for such a scenario, correct?

Reply:
Yes, the test will still fail in such a scenario. This PR only attempts to fix inadvertent substring matches.

Reply:
Can this fix be modified to only match the complete string, for example, Ethernet1 to Ethernet1 but remove the second condition check: "re.search(QUEUE_COUNTERS_RE_FMT.format(setup['port_name_map'][intf]),
                                   queue_counter) is None"

Reply:
I could do this; my worry is that this will remove a check that was intentionally added to the test. If we're confident that the check is unnecessary though I can remove it.